pkg/init/Dockerfile

```diff
@@ -1,3 +1,9 @@
+FROM linuxkit/alpine:630ee558e4869672fae230c78364e367b8ea67a9 AS build
+RUN apk add --no-cache --initdb alpine-baselayout make gcc musl-dev
+
+ADD usermode-helper.c .
+RUN make usermode-helper
```

Comment:
using make's "implicit targets" trick, https://www.gnu.org/software/make/manual/html_node/Implicit-Rules.html

You can do it with any .c file in any directory, assuming that file doesn't need any special compiler or linker args.